lang/cs/Org.Apache.REEF.Client/Multi/MultiRuntimeConfigurationBuilder.cs

```diff
@@ -0,0 +1,249 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Org.Apache.REEF.Client.Multi.Parameters;
+using Org.Apache.REEF.Client.Yarn;
+using Org.Apache.REEF.Client.YARN;
+using Org.Apache.REEF.Client.YARN.HDI;
+using Org.Apache.REEF.Client.YARN.Parameters;
+using Org.Apache.REEF.Common.Runtime;
+using Org.Apache.REEF.IO.FileSystem.AzureBlob;
+using Org.Apache.REEF.Tang.Formats;
+using Org.Apache.REEF.Tang.Implementations.Configuration;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Org.Apache.REEF.Tang.Interface;
+using Org.Apache.REEF.Tang.Util;
+using Org.Apache.REEF.Utilities;
+
+namespace Org.Apache.REEF.Client.Multi
+{
+    public class MultiRuntimeConfigurationBuilder
+    {
+        private const string MultiRuntimeLauncher =
+            "org.apache.reef.bridge.client.MultiRuntimeYarnBootstrapREEFLauncher";
+        private readonly HashSet<RuntimeName> supportedRuntimes = new HashSet<RuntimeName>()
+                                                                      {
+                                                                          RuntimeName.Yarn,
+                                                                          RuntimeName.Local
+                                                                      };
+
+        private HashSet<RuntimeName> runtimes = new HashSet<RuntimeName>();
+
+        private Optional<RuntimeName?> defaultRuntime = Optional<RuntimeName?>.Empty();
```

Comment:
The serialization for string behaves strangely for optional with enum. It generates the first value of enum in the string.